internal/provider/resource_status_page.go

```diff
@@ -0,0 +1,280 @@
+package provider
+
+import (
+	"context"
+	"strconv"
+
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/squadcast/terraform-provider-squadcast/internal/api"
+	"github.com/squadcast/terraform-provider-squadcast/internal/tf"
+)
+
+func resourceStatusPage() *schema.Resource {
+	return &schema.Resource{
+		Description: "Status page resource.",
+
+		CreateContext: resourceStatusPageCreate,
+		ReadContext:   resourceStatusPageRead,
+		UpdateContext: resourceStatusPageUpdate,
+		DeleteContext: resourceStatusPageDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceStatusPageImport,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Description: "Status page id.",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"team_id": {
+				Description:  "Team id.",
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: tf.ValidateObjectID,
+			},
+			"name": {
+				Description:  "Status page name.",
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringLenBetween(1, 1000),
+			},
+			"description": {
+				Description: "Description.",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			"is_public": {
+				Description: "Description.",
+				Type:        schema.TypeBool,
+				Required:    true,
+			},
+			"domain_name": {
+				Description: "Domain name of the status page. This will be appended to https://statuspage.squadcast.com/<ORG_ID>/ to form the URL of the status page.",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"custom_domain_name": {
+				Description: "Custom domain name of the status page.",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			"timezone": {
+				Description: "Timezone",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"contact_email": {
+				Description: "Contact email.",
```

Comment:
A bit more description is needed here. try to pull it from status page feature release doc. or from prakya